### PR TITLE
add/state-in-query-string

### PR DESCRIPTION
**Pull request by ovvost**
Added the ability to use appState with query parameters. 
Some of parameters fill another string and we add this string to one query parameter.
Parameters list in StateConf.queryParamsList
Query parameter name in StateConf.queryParamName

### CODE REVIEW

File: components/appState/index.js [hunk 1]
```diff
@@ -72,6 +72,17 @@ AppState.prototype.getPrevState = function() {
     return this.prevState;
 };
 
+AppState.prototype.getStringToParse = function(url) {
```
Comment (fobos) [inline components/appState/index.js]: JSDoc here

File: lib/stuff.js [hunk 2]
```diff
@@ -165,15 +165,26 @@ exports.state2uri = function(state) {
...
         if (!part) return;
         part.replace(/([^=]+)\=?(.*)/g, function(m, name, value) {
             result[name] = decodeURIComponent(value);
         });
     });
-
     return result;
 };
+
+exports.getLocation = function(url) {
```
Comment (fobos) [inline lib/stuff.js]: JSDoc here

File: components/appState/uri/resolver.js [hunk 3]
```diff
@@ -46,19 +60,18 @@ module.exports = function(stateConf, state, aliases, encode) {
...
-        if (_.isArray(data)) {
-            _.each(data, function(state) {
-                var type = state.type;
-                var data = _.omit(state, 'type');
-                processState(data, type);
-            });
+    var queryParamsList = stateConf.get('queryParamsList') || [];
+    var queryKeys = [];
+    var mainKeys = [];
+    _.each(stateKeys, function(stateKey) {
```
Comment (fobos) [inline components/appState/uri/resolver.js]:
```js
var parts = _.partition(stateKeys, function (key) {
    return _.contains(queryParamsList, key);
});

var pathName = processPart(parts[0]);
var queryPart = processPart(parts[1]);
```

Comment (fobos) [inline components/appState/uri/resolver.js]: I have updated lodash version to 3.10.1 and bump Slot version to 0.12.0. Now we can use all lodash magic ;)

**Comment by fobos** [comment]
looks good :+1:

File: components/appState/uri/resolver.js (hunk 3 above)
Comment (ovvost) [inline components/appState/uri/resolver.js]: @fobos fixed. But I use _.groupBy, it is more readable

**Pull request merged**